### DISCUSSION TITLE
Формальные определния ?

### DISCUSSION THREAD

**Discussion by gochaorg** [twirl/The-API-Book#7] [Ideas]
В книге - глава 2ая, дается неформальное определение - это хорошо, 
но формальное определение устраняет двоякое толкование терминов.

Обычно же дается формальное определение, а потом его разжовывают в неформальном виде, на примерах, третим шагом чтоб сформирвать понимание - резюмируют, то что разжевали так, что бы резюме соответствовало пониманию.

Например глава 9 - используется _Поддержание интероперабельности_ 
Интероперабельность - для новичков мало что скажет, хотя текст выше оринтирован на подрастающее покаление (вроде как, хотя не сказан целевой возраст аудитории), а опытные придерутся.
Та же вики на слово интероперабельность, дает нормальное определение - Функциональная совместимость.

Такая же история с абрагированием - глава 9.
С этим термином есть нормальное определие в учебниках по философии мышления

Вот пример самого термина

> [Абстрагирование](https://ru.wikipedia.org/wiki/%D0%90%D0%B1%D1%81%D1%82%D1%80%D0%B0%D0%BA%D1%86%D0%B8%D1%8F) — отвлечение в процессе познания от несущественных сторон, свойств, связей объекта (предмета или явления) с целью выделения их существенных, закономерных признаков. Результат абстрагирования — абстрактные понятия, например: цвет, кривизна, красота и т. д. 

> [абстрагирование](https://gtmarket.ru/concepts/7009) трактуется как логическая операция поэтапного мысленного членения рассматриваемого объекта или явления, продуцирующая понятия, которые, в свою очередь, образуют всё более общие образы действительности. 

> В процедуре абстрагирования очевидны две ипостаси процесса познания: первая отражает направленность внимания на то, что именно вычленяется; вторая  — на то, от чего именно осуществляется отвлечение.

И далее можно приводить конкретные пример абстрагирования одного от другого

- миниммизация кол-ва необходимых сущьностей - исходя из принципа минимальной достаточности
- функциональная совместимость
- ....

можно конечно и более сильно разогнаться, тут уже от знания философиии, сопоставить задачи абстрагирования философии с проектрированием api

> классификация типов абстракций:

>   1. **изолирующая абстракция** соответствует цели вычленения и чёткой фиксации исследуемого явления;
>   2. **обобщающая абстракция** соответствует цели получения общей картины явления;
>    3. **идеализация** соответствует цели замещения реальной, эмпирически данной ситуации идеализированной схемой для упрощения изучаемой ситуации и более эффективного использования существующих методов и средств исследования.

**Comment by twirl** [comment]
Да, я планирую пройтись и расставить ссылки на рекомендованную литературу по каждому нетривиальному термину.
